GCP Provider panic

**Describe the bug**
Running the `gcp.project.compute.instances { disks {*}}` results in a panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x103cc9cf0]

goroutine 17065 [running]:
go.mondoo.com/cnquery/v10/providers/gcp/resources.(*mqlGcpProjectComputeServiceDisk).MqlID(0x1057cec60?)
        /home/runner/_work/cnquery/cnquery/providers/gcp/resources/gcp.lr.go:11463
go.mondoo.com/cnquery/v10/llx.resource2result({0x1045801a0?, 0x0?}, {0x14001a03840, 0x20})
        /home/runner/_work/cnquery/cnquery/llx/data_conversions.go:321 +0x58
go.mondoo.com/cnquery/v10/llx.raw2primitive({0x1045801a0, 0x0}, {0x14001a03840, 0x20})
        /home/runner/_work/cnquery/cnquery/llx/data_conversions.go:352 +0x14c
go.mondoo.com/cnquery/v10/llx.(*RawData).Result(0x140016ff608)
        /home/runner/_work/cnquery/cnquery/llx/data_conversions.go:377 +0x7c
go.mondoo.com/cnquery/v10/providers-sdk/v1/plugin.(*TValue[...]).ToDataRes(0x103f4c308?, {0x14001a03840?, 0x103d43ec5?})
        /home/runner/_work/cnquery/cnquery/providers-sdk/v1/plugin/runtime.go:214 +0x1b0
go.mondoo.com/cnquery/v10/providers/gcp/resources.init.func1493({0x104636e20?, 0x1400065fb88?})
        /home/runner/_work/cnquery/cnquery/providers/gcp/resources/gcp.lr.go:1256 +0x74
go.mondoo.com/cnquery/v10/providers/gcp/resources.GetData({0x104636e20, 0x1400065fb88}, {0x14001df1630, 0x6}, 0x1400002c1c0?)
        /home/runner/_work/cnquery/cnquery/providers/gcp/resources/gcp.lr.go:4017 +0x130
go.mondoo.com/cnquery/v10/providers-sdk/v1/plugin.(*Service).GetData(0x140022a4928?, 0x1400002c1c0)
        /home/runner/_work/cnquery/cnquery/providers-sdk/v1/plugin/service.go:189 +0x290
go.mondoo.com/cnquery/v10/providers-sdk/v1/plugin.(*GRPCServer).GetData(0x10445bc80?, {0x1400002c1c0?, 0x140022a4978?}, 0x1032ee1e8?)
        /home/runner/_work/cnquery/cnquery/providers-sdk/v1/plugin/grpc.go:129 +0x2c
go.mondoo.com/cnquery/v10/providers-sdk/v1/plugin._ProviderPlugin_GetData_Handler({0x1044002a0, 0x14000b15160}, {0x104641658, 0x14001e716b0}, 0x14002309480, 0x0)
        /home/runner/_work/cnquery/cnquery/providers-sdk/v1/plugin/plugin_grpc.pb.go:299 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000160a00, {0x104641658, 0x14001e71620}, {0x1046495f8, 0x1400063e820}, 0x14001eb3200, 0x14000766030, 0x1057f7cd0, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1386 +0xb58
google.golang.org/grpc.(*Server).handleStream(0x14000160a00, {0x1046495f8, 0x1400063e820}, 0x14001eb3200)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1797 +0xb10
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1027 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 22
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1038 +0x13c
x plugin process exited error="exit status 2" id=36686 plugin=/Users/scottford/.config/mondoo/providers/gcp/gcp
2 errors occurred:
        * 2 errors occurred:
        * the 'gcp' provider crashed: rpc error: code = Unavailable desc = error reading from server: EOF
        * the 'gcp' provider crashed: rpc error: code = Unavailable desc = connection error: desc = "transport: error while dialing: dial unix /var/folders/0h/g13vfqvn2xq7gxj9gdn1z33r0000gn/T/plugin3705972751: connect: connection refused"

        * the 'gcp' provider crashed: rpc error: code = Unavailable desc = connection error: desc = "transport: error while dialing: dial unix /var/folders/0h/g13vfqvn2xq7gxj9gdn1z33r0000gn/T/plugin3705972751: connect: connection refused"
gcp.project.compute.instances: [
  0: {
    disks: 2 errors occurred:
        * the 'gcp' provider crashed: rpc error: code = Unavailable desc = error reading from server: EOF
        * the 'gcp' provider crashed: rpc error: code = Unavailable desc = connection error: desc = "transport: error while dialing: dial unix /var/folders/0h/g13vfqvn2xq7gxj9gdn1z33r0000gn/T/plugin3705972751: connect: connection refused"

  }
  1: {
    disks: the 'gcp' provider crashed: rpc error: code = Unavailable desc = connection error: desc = "transport: error while dialing: dial unix /var/folders/0h/g13vfqvn2xq7gxj9gdn1z33r0000gn/T/plugin3705972751: connect: connection refused"
  }
]
```

--- a/providers/gcp/resources/common.go
+++ b/providers/gcp/resources/common.go
@@ -4,6 +4,7 @@
 package resources
 
 import (
+	"errors"
 	"strings"
 	"time"
 
@@ -105,4 +106,16 @@ func getSubnetworkByUrl(subnetUrl string, runtime *plugin.Runtime) (*mqlGcpProje
 		return nil, err
 	}
 	return res.(*mqlGcpProjectComputeServiceSubnetwork), nil
+}
+
+func getDiskIdByUrl(diskUrl string) (*resourceId, error) {
+	// A reference to a subnetwork is not mandatory for this resource
+	if diskUrl == "" {
+		return nil, errors.New("diskUrl is empty")
+	}
+
+	// Format is https://www.googleapis.com/compute/v1/projects/project1/regions/us-central1/disks/subnet-1
+	params := strings.TrimPrefix(diskUrl, "https://www.googleapis.com/compute/v1/")
+	parts := strings.Split(params, "/")
+	return &resourceId{Project: parts[1], Region: parts[3], Name: parts[5]}, nil
 }

--- a/providers/gcp/resources/compute.go
+++ b/providers/gcp/resources/compute.go
@@ -443,9 +443,31 @@ func (g *mqlGcpProjectComputeServiceAttachedDisk) id() (string, error) {
 }
 
 func (g *mqlGcpProjectComputeServiceAttachedDisk) source() (*mqlGcpProjectComputeServiceDisk, error) {
-	// g.attachedDiskSource
-	// TODO search for reference resource
-	return nil, nil
+	diskId, err := getDiskIdByUrl(g.attachedDiskSource)
+	if err != nil {
+		return nil, err
+	}
+
+	obj, err := CreateResource(g.MqlRuntime, "gcp.project.computeService", map[string]*llx.RawData{
+		"projectId": llx.StringData(g.ProjectId.Data),
+	})
+	if err != nil {
+		return nil, err
+	}
+	computeSvc := obj.(*mqlGcpProjectComputeService)
+	disks := computeSvc.GetDisks()
+	if disks.Error != nil {
+		return nil, disks.Error
+	}
+
+	for _, d := range disks.Data {
+		disk := d.(*mqlGcpProjectComputeServiceDisk)
+		if disk.Zone.Data.GetName().Data == diskId.Region && disk.Name.Data == diskId.Name {
+			return disk, nil
+		}
+
+	}
+	return nil, errors.New("disk not found")
 }
 
 type mqlGcpProjectComputeServiceInstanceInternal struct {

